### PR TITLE
Disable tabs by network configuration

### LINKED ISSUE #1721: Hide unsupported tabs by network
### Description of the Feature

It should be possible to disable or hide some tabs that doesn't make sense in a selected network.

### PR DESCRIPTION
**Pull request by alepc253**
Closes #1721

### Description

Gives the possibility of disable navigation tabs that don't make sense in a certain network.

### Changes

* Addition of new parameter `unsupportedTabs` in network configuration. 
* Addition of new prop `isNotEnabled` in NavigationLink (this prop determines the rendering of the component).

### Steps to Test

1. Choose a network config and add, for example, `unsupportedTabs: ['NAV_ENS', 'NAV_SWAP']` in file `staticNetworks.ts`
2. Select that network
3. Check visually that this tabs are not displayed

### CODE REVIEW

File: common/components/NavigationLink.tsx [hunk 1]
```diff
@@ -8,13 +8,18 @@ interface Props extends RouteComponentProps<{}> {
   link: NavigationLink;
   isHomepage: boolean;
   className: string;
+  isNotEnabled: boolean;
```
Comment (dternyak) [inline common/components/NavigationLink.tsx]: Can we make this an optional boolean? If its not set, it should be falsy anyway.

File: common/components/NavigationLink.tsx [hunk 2]
```diff
@@ -8,13 +8,18 @@ interface Props extends RouteComponentProps<{}> {
...
 }
 
 class NavigationLinkClass extends React.PureComponent<Props, {}> {
   public render() {
-    const { link, location, isHomepage, className } = this.props;
+    const { link, location, isHomepage, className, isNotEnabled } = this.props;
     let isActive = false;
 
+    if (isNotEnabled) {
+      return '';
```
Comment (dternyak) [inline common/components/NavigationLink.tsx]: `return null` is more conventional imho.

File: common/components/Header/components/Navigation.tsx [hunk 3]
```diff
@@ -57,6 +58,7 @@ export default class Navigation extends PureComponent<Props, State> {
                 link={link}
                 isHomepage={link === navigationLinks[0]}
                 className="NavigationLink"
+                isNotEnabled={unsupportedTabs != null && unsupportedTabs.includes(link.name)}
```
Comment (dternyak) [inline common/components/Header/components/Navigation.tsx]: nitpick: `unsupportedTabs && unsupportedTabs.includes(link.name)` is more concise

File: common/components/Header/components/Navigation.tsx [hunk 4]
```diff
@@ -5,6 +5,7 @@ import './Navigation.scss';
 
 interface Props {
   color?: string | false;
+  unsupportedTabs?: string[];
```
Comment (dternyak) [inline common/components/Header/components/Navigation.tsx]: Would prefer to create a union of all tabs and so we can get some type safety around what tabs will be unsupported:
```
type TABS = 'CREATE' | 'SWAP' | 'CONTRACT' ...

const unsupportedTabs: TABS[] = ['SWAP'];
```

Comment (HenryNguyen5) [reply common/components/Header/components/Navigation.tsx]: Would prefer enums here (can be string enums if needed) such as 
```ts
enum TAB {
 CREATE,
 SWAP, 
}
const unsupportedTabs: TAB[] = [TAB.CREATE]
```

**Comment by alepc253** [comment]
@dternyak @HenryNguyen5 thanks for your feedback!
I don't understand what happened but I've lost this branch so I'm creating a new pull request, setting a reference to this one.
Very sorry for the inconvenience.

**Pull request closed without merging**